### PR TITLE
Implement oauth

### PR DESCRIPTION
**Pull request by bheim6**
@JeanJoeris  OAuth working on development, will need to make sure that private keys are updated via figaro after pulling down in order for tests to pass. Currently my keys are being used in stub_omniauth, will need to coordinate when pushing into Production.

### CODE REVIEW

**Comment by bheim6** [comment]
fyi, we shoudl merge this before navbar PR

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -10,6 +10,12 @@
   <body>
     <%= render partial: 'shared/navbar' %>
     <div class="container">
+      <% flash.each do |type, message| %>
```
Comment (JeanJoeris) [inline app/views/layouts/application.html.erb:13]: nice

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  validates_presence_of :name
+  validates_presence_of :token
+
+  def self.from_omniauth(auth_info)
+      user = self.find_or_create_by(uid: auth_info.uid)
+      user.name           = auth_info.info.name
+      user.nickname       = auth_info.info.nickname
+      user.token          = auth_info.credentials.token
+      user.save
+      user
```
Comment (JeanJoeris) [inline app/models/user.rb:11]: I haven't used omniauth before, I like it

File: spec/features/user_can_login_with_github_spec.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    visit '/'
+
+    expect(page).to have_content('Login')
+
+    click_link 'Login'
+
+    expect(current_path).to eq('/')
+    expect(page).to have_content('Hello, Sauron')
+    expect(page).to have_content('Successfully Logged In!')
+    expect(page).to have_content('Logout')
```
Comment (JeanJoeris) [inline spec/features/user_can_login_with_github_spec.rb:29]: hooray for testing

**Review by JeanJoeris — APPROVED** [review]
Looks good, ready to merge

**Pull request merged**